java/pom.xml

```diff
@@ -17,13 +17,39 @@
 
   <properties>
     <project.build.sourceEncoding>UTF-8</project.build.sourceEncoding>
-    <grpc.version>1.16.0</grpc.version>
+    <grpc.version>1.64.0</grpc.version>
     <boringssl.version>2.0.17.Final</boringssl.version>
     <jetty.version>9.2.10.v20150310</jetty.version>
-    <hadoopVersion>2.7.2</hadoopVersion>
+    <protobuf.version>3.25.3</protobuf.version>
+    <hadoopVersion>3.3.5</hadoopVersion>
     <skein.version>UNKNOWN</skein.version>
+    <maven.compiler.source>1.8</maven.compiler.source>
+    <maven.compiler.target>1.8</maven.compiler.target>
   </properties>
 
+  <profiles>
+    <profile>
+        <id>macos</id>
+        <activation>
+            <os>
+                <family>mac</family>
+            </os>
+        </activation>
+        <properties>
+            <archSuffix>:osx-x86_64</archSuffix>
+        </properties>
+    </profile>
```

Comment:
Do we really need this macos profile? If everything is running in Docker wouldn't the environment be Linux, aarch64 or x86_64 and never macos?

Reply:
It's useful to run some of tests locally, but we can remove it.

Reply:
Let's keep it sorry didn't see this change was in the skein fork, I thought this was in hadoop-cluster. I can see it being useful here because we don't necessarily need to always use this package in hadoop-cluster.